Source/Core/DolphinQt/Config/Mapping/MappingCommon.cpp

```diff
@@ -123,6 +124,7 @@ class MappingProcessor : public QWidget
     auto& button = m_clicked_mapping_buttons.front();
     SetButtonExpression(
         button, BuildExpression(detections, default_device, ciface::MappingCommon::Quote::On));
+    m_parent->GetController()->GetConfig()->GenerateControllerTextures();
```

Comment:
There's another `SetButtonExpression` call in this file that didn't get a matching `GenerateControllerTextures` call.
You can just put `GenerateControllerTextures` inside `SetButtonExpression` probably?

Having said that, remembering to stick these calls everywhere is getting messy. It was already bad enough with `UpdateReference`. I haven't super closely examined all the code, but would it make sense to just call `GenerateControllerTextures` inside `UpdateReferences` and `UpdateSingleControlReference` ?

Reply:
Thanks, I don't think `SetButtonExpression` was there when I implemented this or I missed it.  I'll move it there.

> Having said that, remembering to stick these calls everywhere is getting messy. It was already bad enough with UpdateReference.

It has to be done when the controller binding gets updated.  I'm a bit nervous to add it to some global function, as that is what led to the over-generation in the first place.  `UpdateReferences` for instance gets called in a lot of places, in particular it gets called in `LoadConfig()` and that function gets called many times.

Reply:
Texture generation only needs to happen:

* At startup of the title being loaded
* When the profile changes in the UI or via the profile cycling method on Wii
* When a button gets bound in the UI
* Clear / default button gets clicked

It seems clearer from a maintenance standpoint to call out the explicit cases.  So it's easy to see when generation would happen.  Trying to trace through the input system is very challenging for someone who isn't familiar with it.